dts/arm/ti/mspm0/g/mspm0g1106.dtsi

```diff
@@ -0,0 +1,16 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#include <mem.h>
+#include <ti/mspm0/g/mspm0g110x.dtsi>
+
+/ {
+	soc {
+		sram0: memory@20000000 {
+			reg = <0x20000000 DT_SIZE_K(32)>;
+		};
+
+		flash0: serial-flash@0 {
```

Comment:
Difference between xxx6 and xxx7 is the flash size, and thus can be factorized. (perhaps the ram as well)

Refer to what was done there (and only that, this commit brings unrelated stuff like pincrl headers etc..): https://github.com/zephyrproject-rtos/zephyr/commit/0782b116d10140c260d87f6d84cbb589a31befa2

Reply:
In my opinion, creating a shared *.dtsi for memory just because a group of SoCs have similar memory configurations isn't a good idea.